### PR TITLE
Adding Ludwig contributing doc from Ludwig-ai to Ludwig-docs

### PR DESCRIPTION
**Pull request by dalianaliu**

### CODE REVIEW

File: docs/developer_guide/contributing.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Contributing
+
+Everybody is welcome to contribute, and we value everybody’s contribution. Code is thus not the only
+way to help the community. Answering questions, helping others, reaching out and improving the
+documentation are immensely valuable contributions as well.
+
+It also helps us if you spread the word: reference the library from blog posts on the awesome
```
Comment (arnavgarg1) [inline docs/developer_guide/contributing.md:7]: nit: maybe it needs to be changed to `reference the library in blog posts..`?

File: docs/developer_guide/contributing.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+1. Clone your fork to your local disk, and add the base repository as a remote:
+
+   ```bash
+   git clone git@github.com:<your Github handle>/ludwig.git
+   cd ludwig
+   git remote add upstream https://github.com/ludwig-ai/ludwig.git
+   ```
+
+1. Create a new branch to hold your development changes:
+
```
Comment (arnavgarg1) [inline docs/developer_guide/contributing.md:39]: I know that things update correctly when the README is rendered, but are we able to increment these to 2., 3. etc just for better readability?

Comment (arnavgarg1) [reply docs/developer_guide/contributing.md:39]: One more idea for more principled development here is to create a branch from the UI. This creates a link between a branch name and the issue name, which is kind of nice. 

Screen Shot 2022-09-06 at 11 30 11 AM

File: docs/developer_guide/contributing.md (hunk 1 above)
Comment (dalianaliu) [reply docs/developer_guide/contributing.md]: done.

File: docs/developer_guide/contributing.md (hunk 2 above)
Comment (dalianaliu) [reply docs/developer_guide/contributing.md]: > I know that things update correctly when the README is rendered, but are we able to increment these to 2., 3. etc just for better readability?

done.

> One more idea for more principled development here is to create a branch from the UI (bottom right). This creates a link between a branch name and the issue name, which is kind of nice.

I'm not familiar with that workflow, would you be able to add this to doc? Maybe in a separate PR.

Comment (arnavgarg1) [reply docs/developer_guide/contributing.md]: Yup, I can add that as a follow-up PR.

File: docs/developer_guide/contributing.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+   ```
+
+   Push the changes to your account using:
+
+   ```bash
+   git push -u origin a-descriptive-name-for-my-changes
+   ```
+
+8. Once you are satisfied, go the webpage of your fork on GitHub. Click on "Pull request" to send
+   your contribution to the project maintainers for review.
```
Comment (arnavgarg1) [inline docs/developer_guide/contributing.md:86]: Should we add a 9th step to create a PR to update the docs if it is relevant/a new feature?

**Review by justinxzhao — APPROVED** [review]

**Pull request merged**